Source/WebKit/Shared/ios/WebPreferencesDefaultValuesIOS.mm

```diff
@@ -75,36 +75,33 @@ bool defaultMediaSourceEnabled()
 
 #endif
 
-bool defaultUseAsyncUIKitInteractions()
+static bool isAsyncTextInputFeatureFlagEnabled()
 {
     static bool enabled = false;
 #if PLATFORM(IOS) && HAVE(UI_ASYNC_TEXT_INTERACTION)
     static std::once_flag flag;
     std::call_once(flag, [] {
-        enabled = PAL::deviceClassIsSmallScreen() && os_feature_enabled(UIKit, async_text_input);
+        if (PAL::deviceClassIsSmallScreen())
+            enabled = os_feature_enabled(UIKit, async_text_input_iphone) || os_feature_enabled(UIKit, async_text_input);
+        else if (PAL::deviceHasIPadCapability())
+            enabled = os_feature_enabled(UIKit, async_text_input_ipad);
```

Comment:
Wouldn't this disable the feature on iPad for current versions of UIKit in the build?

Reply:
I was thinking the logic here would look more like this:

```cpp
        enabled = os_feature_enabled(UIKit, async_text_input);
        if (PAL::deviceClassIsSmallScreen())
            enabled |= os_feature_enabled(UIKit, async_text_input_iphone);
        else if (PAL::deviceHasIPadCapability())
            enabled |= os_feature_enabled(UIKit, async_text_input_ipad);
```

Reply:
Ah, the intent is exactly to disable the feature on iPad for now (but temporarily keep support for `async_text_input`, as an alias for `async_text_input_iphone`).

Reply:
Oh I see! My bad. Then this is good to go.